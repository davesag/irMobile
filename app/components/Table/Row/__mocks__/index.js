import React from 'react'
import { View, Text } from 'react-native'

export default jest.fn(({ data, makeKey }) => (
  <View>
    {data.map((d, i) => (
      <Text key={makeKey(d, i)}>d</Text>
    ))}
  </View>
))
