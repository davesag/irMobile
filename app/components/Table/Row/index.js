import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { Text } from 'react-native-elements'

import styles from '../styles'

const Row = ({ data, makeKey }) => (
  <View style={styles.row}>
    {data.map((datum, i) => (
      <View key={makeKey(datum, i)} style={styles.dataCell}>
        <Text style={styles.data}>{datum}</Text>
      </View>
    ))}
  </View>
)

Row.propTypes = {
  data: PropTypes.array.isRequired,
  makeKey: PropTypes.func
}

Row.defaultProps = {
  makeKey: (text, index) => `${text}-${index}`
}

export default Row
