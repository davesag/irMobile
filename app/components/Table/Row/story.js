import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { SafeAreaView, View } from 'react-native'

import Row from '.'
import styles from '../styles'

const data = ['Aye', 'Bee', 'Sea', 'Dee']

storiesOf('Table.Row', module).add('with simple data', () => (
  <SafeAreaView>
    <View style={styles.table}>
      <Row data={data} />
    </View>
  </SafeAreaView>
))
