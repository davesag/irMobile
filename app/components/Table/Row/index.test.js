import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Row from '.'

let tree

describe('rendering', () => {
  const data = ['Aye', 'Bee', 'Sea']

  beforeAll(() => {
    tree = renderer.create(<Row data={data} />)
  })

  it('rendered correctly ', () => {
    expect(tree).toMatchSnapshot()
  })
})
