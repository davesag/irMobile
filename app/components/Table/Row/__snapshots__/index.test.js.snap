// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering rendered correctly  1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-evenly",
      "paddingTop": 5,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexWrap": "nowrap",
        "height": 16,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingLeft": 5,
        "paddingRight": 5,
        "width": "25%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexWrap": "nowrap",
        "height": 16,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingLeft": 5,
        "paddingRight": 5,
        "width": "25%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexWrap": "nowrap",
        "height": 16,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingLeft": 5,
        "paddingRight": 5,
        "width": "25%",
      }
    }
  />
</View>
`;
